### PR TITLE
Add statistic for new pool methods

### LINKED ISSUE #443: Count statistics for new pool methods
#434 focuses on making the `Pool` to provide a `Client`-like interface, but this new API is not included in statistics which is important for proper pool node management.

### PR DESCRIPTION
**Pull request by smallhive**
close #443

### CODE REVIEW

**Review by roman-khimov — COMMENTED** [review]
Seems to be a good base for #431 as well.

File: pool/netmap.go [hunk 1]
```diff
@@ -11,22 +12,30 @@ import (
...
-	c, err := p.sdkClient()
+	c, statUpdate, err := p.sdkClient()
 	if err != nil {
 		return netmap.NetMap{}, err
 	}
 
-	return c.NetMapSnapshot(ctx, prm)
+	netMap, err := c.NetMapSnapshot(ctx, prm)
+	statUpdate.updateErrorRate(err)
+
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pool/netmap.go:41]: Extended

File: pool/object.go [hunk 2]
```diff
@@ -161,10 +242,9 @@ func (p *Pool) AnnounceLocalTrust(ctx context.Context, epoch uint64, trusts []re
 //
 // See details in [client.Client.ObjectSearchInit].
 func (p *Pool) AnnounceIntermediateTrust(ctx context.Context, epoch uint64, trust reputation.PeerToPeerTrust, prm client.PrmAnnounceIntermediateTrust) error {
-	c, err := p.sdkClient()
+	c, _, err := p.sdkClient()
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pool/object.go]: Added `updateErrorRate`

File: pool/object.go [hunk 3]
```diff
@@ -161,10 +242,9 @@ func (p *Pool) AnnounceLocalTrust(ctx context.Context, epoch uint64, trusts []re
 //
 // See details in [client.Client.ObjectSearchInit].
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pool/object.go]: Updated

File: pool/object.go [hunk 4]
```diff
@@ -40,119 +41,199 @@ func (p *Pool) ObjectPutInit(ctx context.Context, hdr object.Object, prm client.
...
+	reader, err := c.ObjectSearchInit(ctx, containerID, prm)
+	statUpdate.updateErrorRate(err)
+
+	return reader, err
 }
 
 // AnnounceLocalTrust sends client's trust values to the NeoFS network participants.
 //
 // See details in [client.Client.ObjectSearchInit].
 func (p *Pool) AnnounceLocalTrust(ctx context.Context, epoch uint64, trusts []reputation.Trust, prm client.PrmAnnounceLocalTrust) error {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pool/object.go]: Removed from pool:
- AnnounceLocalTrust
- AnnounceIntermediateTrust
- ContainerAnounceUsedSpace

File: pool/object.go [hunk 5]
```diff
@@ -40,104 +40,182 @@ func (p *Pool) ObjectPutInit(ctx context.Context, hdr object.Object, prm client.
...
+		ctx,
+		c,
+		containerID,
+		p.actualSigner(prm.Signer()),
+		session.VerbObjectSearch,
+		&prm,
+		statUpdater,
+	); err != nil {
 		return nil, fmt.Errorf("session: %w", err)
 	}
```
Comment (roman-khimov) [inline pool/object.go:220]: `statUpdater.updateErrorRate(err)` here?

Comment (roman-khimov) [reply pool/object.go:220]: And a counter as well.

File: pool/object.go [hunk 6]
```diff
@@ -40,104 +40,182 @@ func (p *Pool) ObjectPutInit(ctx context.Context, hdr object.Object, prm client.
...
+		session.VerbObjectRangeHash,
+		&prm,
+		statUpdater,
+	); err != nil {
 		return [][]byte{}, fmt.Errorf("session: %w", err)
 	}
 
-	return c.ObjectHash(ctx, containerID, objectID, prm)
+	hashes, err := c.ObjectHash(ctx, containerID, objectID, prm)
+	statUpdater.updateErrorRate(err)
```
Comment (roman-khimov) [inline pool/object.go:197]: No counter for ObjectHash?

File: pool/netmap.go [hunk 7]
```diff
@@ -11,22 +12,32 @@ import (
 //
 // See details in [client.Client.EndpointInfo].
```
Comment (roman-khimov) [inline pool/netmap.go:13]: BTW, a different thing, but a problem here.

**Review by roman-khimov — COMMENTED** [review]
Please fix `session:` prefix in commits to `pool:` and also describe why we're deleting some methods.

File: pool/object.go (hunk 5 above)
Comment (smallhive) [reply pool/object.go:220]: In this method we pass `statUpdater` inside and it counts error rate inside, when call the network. This `updateErrorRate` calculates only network errors. Inside `withinContainerSession` may be many different errors

Comment (smallhive) [reply pool/object.go:222]: `ObjectSearch` was noone, it is fixed now. I though You mean put counter inside 219 line

File: pool/object.go (hunk 6 above)
Comment (smallhive) [reply pool/object.go:199]: Already we have such counter

File: pool/netmap.go (hunk 7 above)
Comment (smallhive) [reply pool/netmap.go]: updated

**Comment by smallhive** [comment]
> Please fix `session:` prefix in commits to `pool:` and also describe why we're deleting some methods.

Updated

File: pool/object.go (hunk 5 above)
Comment (roman-khimov) [reply pool/object.go:222]: I wasn't able to find a better line for this comment!

File: pool/netmap.go [hunk 8]
```diff
@@ -2,31 +2,42 @@ package pool
...
+	"time"
 
 	"github.com/nspcc-dev/neofs-sdk-go/client"
 	"github.com/nspcc-dev/neofs-sdk-go/netmap"
 )
 
 // NetworkInfo requests information about the NeoFS network of which the remote server is a part.
 //
-// See details in [client.Client.EndpointInfo].
+// See details in [client.Client.NetworkInfo].
```
Comment (roman-khimov) [inline pool/netmap.go:13]: Deserves a separate commit, of course, but OK.

**Review by roman-khimov — APPROVED** [review]
